### PR TITLE
RSDK-3283 - Board tests

### PR DESCRIPTION
**Pull request by clintpurser**
- FakeBoard
- Board Tests (FakeBoard, Server, Client)
- misc Board Client fixes the tests helped identify
- testing port helper function

### CODE REVIEW

File: test/test_utils.dart [hunk 1]
```diff
@@ -0,0 +1 @@
+int generateTestingPortFromName(String name) => 50000 + (name.hashCode % 10000);
```
[the remark this answers is not in the document]
Comment (maximpertsov) [reply test/test_utils.dart:1]: do we actually need this function? why not just hardcode a port per test?

**Review by maximpertsov — APPROVED** [review]

File: test/test_utils.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (clintpurser) [reply test/test_utils.dart:1]: we tossed that around as an idea, I prefer an approach like this because we will have a lot of tests and manually tracking which ports have been used could be prone to errors.

[the remark this answers is not in the document]
Comment (maximpertsov) [reply test/test_utils.dart:1]: cool, as for the name - nothing else comes to mind - maybe it can be shorted to just `generateTestingPort` for brevity's sake?

```suggestion
int generateTestingPort(String name) => 50000 + (name.hashCode % 10000);
```

File: lib/src/components/board/board.dart [hunk 2]
```diff
@@ -13,10 +13,11 @@ class BoardStatus {
...
 
   factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
-    const boardStatus = BoardStatus(<String, int>{}, <String, int>{});
-    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
-    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = (value.value.toInt()));
-    return boardStatus;
+    final Map<String, int> analogs = {};
+    final Map<String, int> digitalInterrupts = {};
+    pbBoardStatus.analogs.forEach((key, value) => analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => digitalInterrupts[key] = (value.value.toInt()));
```
Comment (njooma) [inline lib/src/components/board/board.dart:19]: This is a little cleaner IMO. Thoughts? 
```suggestion
    final Map<String, int> analogs = pbBoardStatus.analogs.map((key, value) => MapEntry(key, value.value));
    final Map<String, int> digitalInterrupts = pbBoardStatus.digitalInterrupts.map((key, value) => MapEntry(key, value.value.toInt()));
```

File: lib/src/components/board/client.dart [hunk 3]
```diff
@@ -31,17 +31,18 @@ class BoardClient extends Board {
   @override
   Future<void> setGpioState(String pin, bool high, {Map<String, dynamic>? extra}) async {
     await _client.setGPIO(SetGPIORequest(name: name, pin: pin, high: high, extra: extra?.toStruct()));
+    return;
```
Comment (njooma) [inline lib/src/components/board/client.dart:34]: Why is this needed?

Comment (clintpurser) [reply lib/src/components/board/client.dart:34]: oh good catch thank you! it is not needed. I added that while debugging and did not mean to include it. 🪓

File: lib/src/components/board/board.dart (hunk 2 above)
Comment (clintpurser) [reply lib/src/components/board/board.dart:19]: yeah I like that, thank you!

**Review by njooma — APPROVED** [review]
